### PR TITLE
cleanup: EphemeralContainers feature gate related codes

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind cleanup
#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/enhancements/issues/277

#### Special notes for your reviewer:
/cc verbs
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by verb — APPROVED** [review]
Thanks for the cleanup! :tada: 

/lgtm

**Review by mrunalp — APPROVED** [review]

File: staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go [hunk 1]
```diff
@@ -444,12 +439,6 @@ func (o *DebugOptions) debugByEphemeralContainer(ctx context.Context, pod *corev
 	pods := o.podClient.Pods(pod.Namespace)
 	result, err := pods.Patch(ctx, pod.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{}, "ephemeralcontainers")
 	if err != nil {
-		// The apiserver will return a 404 when the EphemeralContainers feature is disabled because the `/ephemeralcontainers` subresource
-		// is missing. Unlike the 404 returned by a missing pod, the status details will be empty.
-		if serr, ok := err.(*errors.StatusError); ok && serr.Status().Reason == metav1.StatusReasonNotFound && serr.ErrStatus.Details.Name == "" {
```
Comment (liggitt) [inline staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go:449]: I'd recommend keeping this code in place to give a helpful error on older servers

**Pull request merged**